tests/console/firewalld.pm

```diff
@@ -0,0 +1,142 @@
+    assert_script_run("iptables -t nat -C PRE_public_allow -p tcp -m mark --mark `iptables -t mangle -L PRE_public_allow | grep -i MARK | sed 's/set /\$ /' | cut -f 2 -d \"\$\"` -j DNAT --to :22");
+
+    # Reload default state and flush rules
+    record_info 'Flushing Rules';
+    assert_script_run("firewall-cmd --reload");
+    assert_script_run("if [ `iptables -t mangle -L PRE_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq 0 ]; then /usr/bin/true; else /usr/bin/false; fi");
+    assert_script_run("if [ `iptables -t nat -L PRE_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq 0 ]; then /usr/bin/true; else /usr/bin/false; fi");
+
+
+    # Test #5 - ipv4 adress family with rich rules
+    record_info 'Test #5", "Test ipv4 family addresses with rich rules';
+    assert_script_run("firewall-cmd --zone=public --permanent --add-rich-rule 'rule family=\"ipv4\" source address=192.168.200.0/24 accept'");
+    assert_script_run("firewall-cmd --zone=public --permanent --add-rich-rule 'rule family=\"ipv4\" source address=192.168.201.0/24 drop'");
+    assert_script_run("firewall-cmd --reload");
+    assert_script_run("iptables -C IN_public_allow -s 192.168.200.0/24 -j ACCEPT");
+    assert_script_run("iptables -C IN_public_deny -s 192.168.201.0/24 -j DROP");
+
+
+    # Reload default state and flush rules
+    record_info 'Flushing rules';
+    assert_script_run("firewall-cmd --zone=public --permanent --remove-rich-rule 'rule family=\"ipv4\" source address=192.168.200.0/24 accept'");
+    assert_script_run("firewall-cmd --zone=public --permanent --remove-rich-rule 'rule family=\"ipv4\" source address=192.168.201.0/24 drop'");
+    assert_script_run("firewall-cmd --reload");
+    assert_script_run("if [ `iptables -L IN_public_allow --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq 0 ]; then /usr/bin/true; else /usr/bin/false; fi");
+    assert_script_run("if [ `iptables -L IN_public_deny --line-numbers | sed '/^num\\|^\$\\|^Chain/d' | wc -l` -eq 0 ]; then /usr/bin/true; else /usr/bin/false; fi");
+
+
+    # Test #6 - Change the default zone
+    record_info 'Test #6', 'Change the default zone';
+    assert_script_run("firewall-cmd --set-default-zone=public");
+    assert_script_run("if [ a`firewall-cmd --get-default-zone | grep -i 'public'` = \"apublic\" ]; then true; else false; fi");
+    assert_script_run("firewall-cmd --set-default-zone=dmz");
+    assert_script_run("if [ a`firewall-cmd --get-default-zone | grep -i 'dmz'` = \"admz\" ]; then true; else false; fi");
+
+    # Change to the default zone
+    record_info 'Set Default Zone';
+    assert_script_run("firewall-cmd --set-default-zone=public");
+    assert_script_run("if [ a`firewall-cmd --get-default-zone | grep -i 'public'` = \"apublic\" ]; then true; else false; fi");
```

Comment:
`firewall-cmd --get-default-zone | grep -q  -i 'public'` instead `if then else` construction is sufficient grep returns 0 if is  pattern found else 1

+ same thing for other commands